v1/ao/internal/reporter/reporter_grpc.go

```diff
@@ -290,7 +292,7 @@ func newGRPCReporter() reporter {
 		getSettingsInterval:          grpcGetSettingsIntervalDefault,
 		settingsTimeoutCheckInterval: grpcSettingsTimeoutCheckIntervalDefault,
 
-		serviceKey: config.GetServiceKey(),
+		serviceKey: uatomic.NewString(config.GetServiceKey()),
```

Comment:
What is the expectation on when the service key can be set on the fly?

Based on the description. this will only get called once on program startup, so if this is invoked before the service key is set, is that a valid use case? :)

Reply:
Yes, if it's invoked before the service key is set, all the RPC calls will be with an empty service key and finally get silently skipped. Each time when a RPC call is invoked it will try to get the updated service key from this field `r.serviceKey`.

Reply:
Ah got it thanks